bindings/rust/build.rs

```diff
@@ -196,38 +192,71 @@ fn target_os_is(name: &str) -> bool {
     }
 }
 
-fn target_dylib_extension() -> &'static str {
-    if target_os_is("macos") {
-        ".dylib"
+/// Return an iterator over possible names for the llvm-config binary.
+fn llvm_config_binary_names() -> impl Iterator<Item = String> {
+    let base_names = [
+        "llvm-config".into(),
+        format!("llvm-config-{}", 18),
+        format!("llvm-config{}", 18),
+        format!("llvm{}-config", 18),
```

Comment:
If the format! args refer to the LLVM version 18, it may make sense to add the other supported versions as well.